agent/eni/watcher/watcher_linux.go

```diff
@@ -147,16 +147,26 @@ func (udevWatcher *UdevWatcher) reconcileOnce() error {
 // sendENIStateChange handles the eni event from udev or reconcile phase
 func (udevWatcher *UdevWatcher) sendENIStateChange(mac string) {
 	eniAttachment, ok := udevWatcher.shouldSendENIStateChange(mac)
-	if ok {
-		go func(eni *api.ENIAttachment) {
-			eni.Status = api.ENIAttached
-			log.Infof("Emitting ENI change event for: %v", eni)
-			udevWatcher.eniChangeEvent <- api.TaskStateChange{
-				TaskARN:    eni.TaskARN,
-				Attachment: eni,
-			}
-		}(eniAttachment)
+	if !ok {
+		if eniAttachment != nil {
+			// Agent is aware of the ENI, but we decide not to ack it
+			// as it's either already been ack'd or because it's ack
+			// timeout has expired
+			log.Infof("ENI state manager: removing ENI attachment from state, no longer tracking it: %s",
+				eniAttachment.String())
+			udevWatcher.agentState.RemoveENIAttachment(eniAttachment.MACAddress)
```

Comment:
why do we remove it if it has already been ack'd?

Reply:
We are discussing that here: https://github.com/aws/amazon-ecs-agent/pull/1005/files/a1ff0d886dbf615a68e1c6b29cbb877fd4fb9189#diff-b92a021259580671251a2c3e099e925c